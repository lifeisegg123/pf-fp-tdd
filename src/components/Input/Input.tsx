import { ChangeEventHandler } from "react";
import { ComponentBaseProps } from "src/interfaces";

interface InputProps extends ComponentBaseProps {
  onChange: ChangeEventHandler<HTMLInputElement>;
  value: string;
  placeholder?: string;
  disabled?: boolean;
}

export function Input(props: InputProps) {
  return <input {...props} />;
}
