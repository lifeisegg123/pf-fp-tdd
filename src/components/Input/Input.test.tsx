import { Input } from "./Input";
import { render, fireEvent } from "@testing-library/react";
import { useState } from "react";

function MockInput({ disabled }: { disabled?: boolean }) {
  const [value, setValue] = useState("");
  return (
    <Input
      placeholder="mock-input"
      value={value}
      disabled={disabled}
      onChange={(e) => setValue(e.target.value)}
    />
  );
}

describe("Component/Input", () => {
  it("should match snapshot", () => {
    const { container } = render(<MockInput />);
    expect(container).toBeInTheDocument();
  });

  it("should handle change event", () => {
    const { getByPlaceholderText } = render(<MockInput />);
    const input = getByPlaceholderText("mock-input") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "test" } });
    expect(input.value).toEqual("test");
  });

  it("should be disabled", () => {
    const { getByPlaceholderText } = render(<MockInput disabled />);
    const input = getByPlaceholderText("mock-input") as HTMLInputElement;
    expect(input.disabled).toEqual(true);
  });
});
