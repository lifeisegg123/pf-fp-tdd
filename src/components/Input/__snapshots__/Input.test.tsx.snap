// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/Input should match snapshot 1`] = `
<div>
  <input
    placeholder="mock-input"
    value=""
  />
</div>
`;
